xde/src/dev_map.rs

```diff
@@ -0,0 +1,90 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+// Copyright 2024 Oxide Computer Company
+
+use crate::xde::XdeDev;
+use alloc::boxed::Box;
+use alloc::collections::btree_map::BTreeMap;
+use alloc::string::String;
+use opte::api::MacAddr;
+use opte::api::Vni;
+
+// From microbenchmarking (https://github.com/oxidecomputer/opte/issues/637)
+// it is apparent that we have *far* faster `Ord` and `Eq` implementations
```

Comment:
hmm, worth updating those impls at some point then?

Reply:
Thanks for the review, Luqman!

I tested this out moving the conversions into the Eq/Ord impls just there – this didn't make any of the `(Vni, MacAddr)` cases as fast as the `(u32, u64)` ones. Linear scans got a bit slower.

[I put a quick proxy of this into godbolt](https://godbolt.org/z/3P99Kx3GE) to see what rustc is actually doing -- it looks to me as though for `MacAddr`, `Ord` works out using `core::intrinsics::compare_bytes` and doing the same BE conversion on the first 4B and last 2B. I think most of the improvement I get in the benchmarks comes down to precaching this conversion for the entire map.

This has me wondering if we can throw the same trick at `InnerFlowId`s -- if there's anything there I'll open a followup.

Reply:
Sadly, I did not see any measurable improvement on packet processing time when trying related shenanigans with `InnerFlowId` -> `[u64; 5]` for `FlowTable`s. This remained the case even at fairly high UFT occupancy.